### PR TITLE
executor: migrate test-infra to testify for index_merge_reader_test.go

### LINKED ISSUE #28586: migrate test-infra to testify for `index_merge_reader_test.go`

### PR DESCRIPTION
**Pull request by hawkingrei**
Signed-off-by: Weizhen Wang <wangweizhen@pingcap.com>

### What problem does this PR solve?

Issue Number: close #28586

Problem Summary:

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: executor/index_merge_reader_test.go [hunk 1]
```diff
@@ -441,22 +465,24 @@ func (test *testSerialSuite2) TestIndexMergeReaderMemTracker(c *C) {
 	tk.MustQuery("select /*+ use_index_merge(t1) */ * from t1 where c1 > 1 or c2 > 1")
 
 	newMaxUsage := memTracker.MaxConsumed()
-	c.Assert(newMaxUsage, Greater, oriMaxUsage)
+	require.Greater(t, newMaxUsage, oriMaxUsage)
 
 	res := tk.MustQuery("explain analyze select /*+ use_index_merge(t1) */ * from t1 where c1 > 1 or c2 > 1")
-	c.Assert(len(res.Rows()), Equals, 4)
+	require.Equal(t, 4, len(res.Rows()))
```
Comment (tisonkun) [inline executor/index_merge_reader_test.go:471]:
```suggestion
	require.Len(t, res.Rows(), 4)
```

**Review by tisonkun — APPROVED** [review]
LGTM. One style comment inline.

**Review by disksing — APPROVED** [review]

**Pull request merged**